plugins/BEdita/Core/src/Model/Entity/Stream.php

```diff
@@ -276,11 +284,13 @@ protected function readFileMetadata($resource): void
     {
         if (preg_match('/image\//', $this->mime_type) && function_exists('getimagesizefromstring')) {
             rewind($resource);
-            $size = getimagesizefromstring(stream_get_contents($resource));
-
-            if (!empty($size)) {
-                $this->width = $size[0];
-                $this->height = $size[1];
+            $content = stream_get_contents($resource);
```

Comment:
This loads the entire image contents in memory as a string. For very large images, this may get the process killed.
Furthermore, it is really done only to obtain image size.

I'd strongly advise revisiting this part. EXIF already returns tons of metadata which includes image size, so perhaps it'd be wiser to use that instead of this approach. It would also have the benefit of reading the stream only once rather than twice.

Reply:
I did some test with various types of image and found out that png, bmp and webp all errored out when using `exif_read_data`.
It seems like only jpeg/jpg and tiff images have EXIF data, so I changed the code to use `getimagesizefromstring` or `exif_read_data` depending on the MIME type.